bundles/model/org.eclipse.smarthome.model.rule.runtime/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleTriggerManager.java

```diff
@@ -574,7 +574,9 @@ public void removeRule(TriggerTypes type, Rule rule) {
                 }
                 break;
             case TRIGGER:
-                triggerEventTriggeredRules.remove(rule);
+                for (Set<Rule> rules : triggerEventTriggeredRules.values()) {
+                    rules.remove(rule);
```

Comment:
I don't see any problem here. He iterates over a collection of set of rules and removed something from the rules.
He does not remove something from the collection he iterates.